From efb3cd4a13bc5b50f2b8ba3138044a9dbdb6bfdc Mon Sep 17 00:00:00 2001
From: dianlujitao <dianlujitao@lineageos.org>
Date: Sat, 23 Feb 2019 20:24:57 +0800
Subject: [PATCH 16/18] power: Pass NULL parameter in powerHint if data is zero

 * This restores the behavior in AOSP and CAF power HAL to avoid
   confusion.

Change-Id: I72f5bb9286e2f57121e39eea82d2fe8854989393
---
 Power.cpp    | 2 +-
 power-660.c  | 5 +++--
 power-8084.c | 5 +++--
 power-8226.c | 5 +++--
 power-8610.c | 5 +++--
 power-8916.c | 5 +++--
 power-8937.c | 5 +++--
 power-8952.c | 5 +++--
 power-8953.c | 5 +++--
 power-8960.c | 5 +++--
 power-8974.c | 5 +++--
 power-8992.c | 5 +++--
 power-8994.c | 5 +++--
 power-8996.c | 5 +++--
 power-8998.c | 5 +++--
 15 files changed, 43 insertions(+), 29 deletions(-)

diff --git a/Power.cpp b/Power.cpp
index 0bf3d69..0765378 100644
--- a/Power.cpp
+++ b/Power.cpp
@@ -55,7 +55,7 @@ Return<void> Power::setInteractive(bool interactive)  {
 }
 
 Return<void> Power::powerHint(PowerHint hint, int32_t data) {
-    power_hint(static_cast<power_hint_t>(hint), &data);
+    power_hint(static_cast<power_hint_t>(hint), data ? (&data) : NULL);
     return Void();
 }
 
diff --git a/power-660.c b/power-660.c
index a866b5b..f82efff 100644
--- a/power-660.c
+++ b/power-660.c
@@ -98,8 +98,9 @@ int get_number_of_profiles()
 }
 #endif
 
-static int set_power_profile(int profile)
+static int set_power_profile(void *data)
 {
+    int profile = data ? *((int*)data) : 0;
     int ret = -EINVAL;
     const char *profile_name = NULL;
 
@@ -288,7 +289,7 @@ static void process_interaction_hint(void *data)
 int power_hint_override(power_hint_t hint, void *data)
 {
     if (hint == POWER_HINT_SET_PROFILE) {
-        if (set_power_profile(*(int32_t *)data) < 0)
+        if (set_power_profile(data) < 0)
             ALOGE("Setting power profile failed. perf HAL not started?");
         return HINT_HANDLED;
     }
diff --git a/power-8084.c b/power-8084.c
index 7d4bbe4..52d80c3 100644
--- a/power-8084.c
+++ b/power-8084.c
@@ -100,8 +100,9 @@ int get_number_of_profiles()
 }
 #endif
 
-static int set_power_profile(int profile)
+static int set_power_profile(void *data)
 {
+    int profile = data ? *((int*)data) : 0;
     int ret = -EINVAL;
     const char *profile_name = NULL;
 
@@ -188,7 +189,7 @@ int power_hint_override(power_hint_t hint, void *data)
     int duration;
 
     if (hint == POWER_HINT_SET_PROFILE) {
-        if (set_power_profile(*(int32_t *)data) < 0)
+        if (set_power_profile(data) < 0)
             ALOGE("Setting power profile failed. mpdecision not started?");
         return HINT_HANDLED;
     }
diff --git a/power-8226.c b/power-8226.c
index 573433a..8a0e0e7 100644
--- a/power-8226.c
+++ b/power-8226.c
@@ -75,8 +75,9 @@ int get_number_of_profiles()
 }
 #endif
 
-static int set_power_profile(int profile)
+static int set_power_profile(void *data)
 {
+    int profile = data ? *((int*)data) : 0;
     int ret = -EINVAL;
     const char *profile_name = NULL;
 
@@ -128,7 +129,7 @@ int power_hint_override(power_hint_t hint, void *data)
     int duration;
 
     if (hint == POWER_HINT_SET_PROFILE) {
-        if (set_power_profile(*(int32_t *)data) < 0)
+        if (set_power_profile(data) < 0)
             ALOGE("Setting power profile failed. mpdecision not started?");
         return HINT_HANDLED;
     }
diff --git a/power-8610.c b/power-8610.c
index 80a1f49..e4f6a6c 100644
--- a/power-8610.c
+++ b/power-8610.c
@@ -71,8 +71,9 @@ int get_number_of_profiles()
 }
 #endif
 
-static int set_power_profile(int profile)
+static int set_power_profile(void *data)
 {
+    int profile = data ? *((int*)data) : 0;
     int ret = -EINVAL;
     const char *profile_name = NULL;
 
@@ -124,7 +125,7 @@ int power_hint_override(power_hint_t hint, void *data)
     int duration;
 
     if (hint == POWER_HINT_SET_PROFILE) {
-        if (set_power_profile(*(int32_t *)data) < 0)
+        if (set_power_profile(data) < 0)
             ALOGE("Setting power profile failed. mpdecision not started?");
         return HINT_HANDLED;
     }
diff --git a/power-8916.c b/power-8916.c
index 204d016..c355290 100644
--- a/power-8916.c
+++ b/power-8916.c
@@ -108,8 +108,9 @@ int get_number_of_profiles()
 }
 #endif
 
-static int set_power_profile(int profile)
+static int set_power_profile(void *data)
 {
+    int profile = data ? *((int*)data) : 0;
     int ret = -EINVAL;
     const char *profile_name = NULL;
 
@@ -185,7 +186,7 @@ int power_hint_override(power_hint_t hint, void *data)
     int duration;
 
     if (hint == POWER_HINT_SET_PROFILE) {
-        if (set_power_profile(*(int32_t *)data) < 0)
+        if (set_power_profile(data) < 0)
             ALOGE("Setting power profile failed. perfd not started?");
         return HINT_HANDLED;
     }
diff --git a/power-8937.c b/power-8937.c
index da3d0e9..75b8395 100644
--- a/power-8937.c
+++ b/power-8937.c
@@ -93,8 +93,9 @@ int get_number_of_profiles()
 }
 #endif
 
-static int set_power_profile(int profile)
+static int set_power_profile(void *data)
 {
+    int profile = data ? *((int*)data) : 0;
     int ret = -EINVAL;
     const char *profile_name = NULL;
 
@@ -233,7 +234,7 @@ static void process_interaction_hint(void *data)
 int power_hint_override(power_hint_t hint, void *data)
 {
     if (hint == POWER_HINT_SET_PROFILE) {
-        if (set_power_profile(*(int32_t *)data) < 0)
+        if (set_power_profile(data) < 0)
             ALOGE("Setting power profile failed. perfd not started?");
         return HINT_HANDLED;
     }
diff --git a/power-8952.c b/power-8952.c
index 521771d..a413d20 100644
--- a/power-8952.c
+++ b/power-8952.c
@@ -91,8 +91,9 @@ int get_number_of_profiles()
 }
 #endif
 
-static int set_power_profile(int profile)
+static int set_power_profile(void *data)
 {
+    int profile = data ? *((int*)data) : 0;
     int ret = -EINVAL;
     const char *profile_name = NULL;
 
@@ -281,7 +282,7 @@ static void process_video_encode_hint(void *metadata)
 int power_hint_override(power_hint_t hint, void *data)
 {
     if (hint == POWER_HINT_SET_PROFILE) {
-        if (set_power_profile(*(int32_t *)data) < 0)
+        if (set_power_profile(data) < 0)
             ALOGE("Setting power profile failed. perfd not started?");
         return HINT_HANDLED;
     }
diff --git a/power-8953.c b/power-8953.c
index bd9211c..873a90b 100644
--- a/power-8953.c
+++ b/power-8953.c
@@ -99,8 +99,9 @@ int get_number_of_profiles()
 }
 #endif
 
-static int set_power_profile(int profile)
+static int set_power_profile(void *data)
 {
+    int profile = data ? *((int*)data) : 0;
     int ret = -EINVAL;
     const char *profile_name = NULL;
 
@@ -335,7 +336,7 @@ int power_hint_override(power_hint_t hint, void *data)
     int ret_val = HINT_NONE;
 
     if (hint == POWER_HINT_SET_PROFILE) {
-        if (set_power_profile(*(int32_t *)data) < 0)
+        if (set_power_profile(data) < 0)
             ALOGE("Setting power profile failed. perf HAL not started?");
         return HINT_HANDLED;
     }
diff --git a/power-8960.c b/power-8960.c
index 495e196..0076cbb 100644
--- a/power-8960.c
+++ b/power-8960.c
@@ -91,8 +91,9 @@ int get_number_of_profiles()
 }
 #endif
 
-static int set_power_profile(int profile)
+static int set_power_profile(void *data)
 {
+    int profile = data ? *((int*)data) : 0;
     int ret = -EINVAL;
     const char *profile_name = NULL;
 
@@ -138,7 +139,7 @@ static int set_power_profile(int profile)
 int power_hint_override(power_hint_t hint, void *data)
 {
     if (hint == POWER_HINT_SET_PROFILE) {
-        if (set_power_profile(*(int32_t *)data) < 0)
+        if (set_power_profile(data) < 0)
             ALOGE("Setting power profile failed. perfd not started?");
         return HINT_HANDLED;
     }
diff --git a/power-8974.c b/power-8974.c
index 94d4125..50df288 100644
--- a/power-8974.c
+++ b/power-8974.c
@@ -111,8 +111,9 @@ int get_number_of_profiles()
 }
 #endif
 
-static int set_power_profile(int profile)
+static int set_power_profile(void *data)
 {
+    int profile = data ? *((int*)data) : 0;
     int ret = -EINVAL;
     const char *profile_name = NULL;
 
@@ -191,7 +192,7 @@ int power_hint_override(power_hint_t hint, void *data)
     int duration;
 
     if (hint == POWER_HINT_SET_PROFILE) {
-        if (set_power_profile(*(int32_t *)data) < 0)
+        if (set_power_profile(data) < 0)
             ALOGE("Setting power profile failed. mpdecision not started?");
         return HINT_HANDLED;
     }
diff --git a/power-8992.c b/power-8992.c
index b6c3413..c267628 100644
--- a/power-8992.c
+++ b/power-8992.c
@@ -96,8 +96,9 @@ int get_number_of_profiles()
 }
 #endif
 
-static int set_power_profile(int profile)
+static int set_power_profile(void *data)
 {
+    int profile = data ? *((int*)data) : 0;
     int ret = -EINVAL;
     const char *profile_name = NULL;
 
@@ -214,7 +215,7 @@ int power_hint_override(power_hint_t hint, void *data)
     int duration;
 
     if (hint == POWER_HINT_SET_PROFILE) {
-        if (set_power_profile(*(int32_t *)data) < 0)
+        if (set_power_profile(data) < 0)
             ALOGE("Setting power profile failed. perfd not started?");
         return HINT_HANDLED;
     }
diff --git a/power-8994.c b/power-8994.c
index f56a873..96f96f7 100644
--- a/power-8994.c
+++ b/power-8994.c
@@ -104,8 +104,9 @@ int get_number_of_profiles()
 }
 #endif
 
-static int set_power_profile(int profile)
+static int set_power_profile(void *data)
 {
+    int profile = data ? *((int*)data) : 0;
     int ret = -EINVAL;
     const char *profile_name = NULL;
 
@@ -222,7 +223,7 @@ int power_hint_override(power_hint_t hint, void *data)
     int duration;
 
     if (hint == POWER_HINT_SET_PROFILE) {
-        if (set_power_profile(*(int32_t *)data) < 0)
+        if (set_power_profile(data) < 0)
             ALOGE("Setting power profile failed. perfd not started?");
         return HINT_HANDLED;
     }
diff --git a/power-8996.c b/power-8996.c
index 38ca714..0b33497 100644
--- a/power-8996.c
+++ b/power-8996.c
@@ -91,8 +91,9 @@ int get_number_of_profiles()
 }
 #endif
 
-static int set_power_profile(int profile)
+static int set_power_profile(void *data)
 {
+    int profile = data ? *((int*)data) : 0;
     int ret = -EINVAL;
     const char *profile_name = NULL;
 
@@ -321,7 +322,7 @@ int power_hint_override(power_hint_t hint, void *data)
     int ret_val = HINT_NONE;
 
     if (hint == POWER_HINT_SET_PROFILE) {
-        if (set_power_profile(*(int32_t *)data) < 0)
+        if (set_power_profile(data) < 0)
             ALOGE("Setting power profile failed. perf HAL not started?");
         return HINT_HANDLED;
     }
diff --git a/power-8998.c b/power-8998.c
index 123c179..6fdffff 100644
--- a/power-8998.c
+++ b/power-8998.c
@@ -87,8 +87,9 @@ int get_number_of_profiles()
 }
 #endif
 
-static int set_power_profile(int profile)
+static int set_power_profile(void *data)
 {
+    int profile = data ? *((int*)data) : 0;
     int ret = -EINVAL;
     const char *profile_name = NULL;
 
@@ -317,7 +318,7 @@ int power_hint_override(power_hint_t hint, void *data)
     int ret_val = HINT_NONE;
 
     if (hint == POWER_HINT_SET_PROFILE) {
-        if (set_power_profile(*(int32_t *)data) < 0)
+        if (set_power_profile(data) < 0)
             ALOGE("Setting power profile failed. perf HAL not started?");
         return HINT_HANDLED;
     }
-- 
2.17.1

